File: permissionlib/src/main/java/com/benyanyi/permissionlib/PermissionDialogInfo.java
package com.benyanyi.permissionlib;

import android.content.DialogInterface;

/**
 * @author myLove
 * @date 2018-06-09 10:02
 * @email ben@yanyi.red
 * @overview
 */
public final class PermissionDialogInfo {
    public String title;
    public String message;
    public String positiveText;
    public String negativeText;

    public boolean isShow = true;

    public PositiveClick positiveClick;
    public NegativeClick negativeClick;

    public interface PositiveClick extends PermissionDialogOnClick {
        /**
         * 点击
         *
         * @param failurePermissions
         */
        @Override
        void onClick(DialogInterface dialog, String[] failurePermissions);
    }

    public interface NegativeClick extends PermissionDialogOnClick {
        /**
         * 点击
         *
         * @param failurePermissions
         */
        @Override
        void onClick(DialogInterface dialog, String[] failurePermissions);
    }

}

File: permission-kt/src/main/java/com/benyanyi/permission/kt/info/PermissionDialogInfo.kt
package com.benyanyi.permission.kt.info

import android.content.DialogInterface

/**
 * @author myLove
 * @date 2018-06-09 10:02
 * @email ben@yanyi.red
 * @overview
 */
class PermissionDialogInfo {
    var title: String? = null
    var message: String? = null
    var positiveText: String? = null
    var negativeText: String? = null
    var isShow = true
    var positiveClick: PositiveClick? = null
    var negativeClick: NegativeClick? = null

    interface PositiveClick : PermissionDialogOnClick {
        /**
         * 点击
         *
         * @param failurePermissions
         */
        override fun onClick(dialog: DialogInterface, failurePermissions: Array<String>)
    }

    interface NegativeClick : PermissionDialogOnClick {
        /**
         * 点击
         *
         * @param failurePermissions
         */
        override fun onClick(dialog: DialogInterface, failurePermissions: Array<String>)
    }
}